datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -828,7 +828,8 @@ mod tests {
             None,
             schema.as_ref(),
         );
-        assert!(expr.is_err());
+        // expr is ok since boolean coercion is supported
```

Comment:
I think we lost the negative coverage here -- can. you please change the test back to checking that this is an error?

Perhaps you could change it to 

```
CASE WHEN a = 'foo' THEN timestamp(500ns) WHEN a = 'bar' THEN true END
```

As bool --> timestamp coercion isn't supported